src/views/language/LanguageTable.tsx

```diff
@@ -14,8 +14,55 @@ import PopulationWarning from '../common/PopulationWarning';
 import { CodeColumn, EndonymColumn, NameColumn } from '../common/table/CommonColumns';
 import ObjectTable from '../common/table/ObjectTable';
 
+import {
+  computeVitalityMetascore,
+  getAllVitalityScores,
+  getVitalityExplanation,
+  VitalityMeterType,
+} from './LanguageVitalityComputation';
+
 const LanguageTable: React.FC = () => {
   const { languagesInSelectedSource } = useDataContext();
+
+  // Helper functions for vitality display
+  function scoreBucket(score: number | null): 0 | 1 | 2 | 3 {
+    if (score === null) return 0;
+    if (score >= 7) return 3; // strong
+    if (score >= 4) return 2; // medium
+    if (score >= 1) return 1; // low
+    return 0; // none/extinct
+  }
+
+  function bucketIcon(bucket: 0 | 1 | 2 | 3) {
+    switch (bucket) {
+      case 3:
+        return <WifiHigh size="1em" />;
+      case 2:
+        return <Wifi size="1em" />;
```

Comment:
Oddly, "WifiHigh" is only 3 bars but "Wifi" is 4 bars. So you should swap them.

Also, the Lucide library offers 2 aliases for the icons, eg. `Wifi` and `WifiIcon`. I recommend using the Icon ones so its clearer what it is.

```suggestion
        return <WifiIcon size="1em" />;
      case 2:
        return <WifiHighIcon size="1em" />;
```